packages/image-block/src/hooks/useImageContainer.tsx

```diff
@@ -0,0 +1,39 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { debounce } from '@frontify/fondue';
+import { useEffect, useRef, useState } from 'react';
+
+export const useImageContainer = () => {
+    const containerRef = useRef<HTMLElement | null>(null);
+    const [containerWidth, setContainerWidth] = useState<number | undefined>(undefined);
+
+    useEffect(() => {
+        if (!containerRef.current) {
+            return;
+        }
+
+        const containerObserver = new ResizeObserver(
+            debounce((entries) => {
+                const container = entries[0];
+                const newContainerWidth = container.contentRect.width;
+                const oldContainerWidth = containerWidth ?? 0;
+                const containerWidthHasGrown = oldContainerWidth < newContainerWidth;
+                if (containerWidthHasGrown) {
+                    setContainerWidth(container.contentRect.width);
+                }
+            }, 100)
+        );
```

Comment:
My only concern with this is that it is a lot harder to cache images. I think it makes more sense to use the pre-determined array of sizes approach, although I'm unaware if there are any rendering issues if for example you render a 1080px image in an 800px container.

Reply:
This would also mean changing the name of the hook probably

Reply:
Yeah that makes sense. To make it flexible, I made it that the container width is always rounded to the next hundred. So possible values are 100, 200, 300, 400, etc...

What do you think?

Reply:
Honestly I think this will come from trial and error to find a nice in-between with image size and number of requests, but for now it works really well